### ISSUE TITLE
Expedite type map refresh after cluster membership change

### ISSUE THREAD

**Issue by ReubenBond** [dotnet/orleans#5178]
When a silo joins a cluster, the cluster type map on other silos is not refreshed immediately.

This means that there is a delay between a new silo joining and the time that other silos begin placing grains on that new silo.

Emile van Reenen inquired about this on Gitter: https://gitter.im/dotnet/orleans?at=5bed7ef392b7d11721924cd5

**Comment by emilevr** [comment]
I've tested using the AssumeHomogenousSilosForTesting property on SiloMessagingOptions, as suggested by @sergeybykov and can confirm that setting that property to true does remove the delay I was seeing during rolling upgrades.

I've also tested without using the flag and in those cases it takes about 45 seconds after the silo becomes _active_ before it will take/receive grain placements.

**Comment by sergeybykov** [comment]
@emilevr Thanks for confirming.

It is by design that it takes _some_ time to get the type maps exchanged. It is surprising that it takes that long.

**Comment by benjaminpetit** [comment]
The default refresh timeout is 1 minute. But you can customize it with `TypeManagementOptions.TypeMapRefreshInterval`

**Comment by emilevr** [comment]
Thanks for the info. Tested it and yes, that does enable me to configure the behaviour to better suit my needs. So it looks like it takes about 10 seconds from starting the silo till it becomes active in the cluster (must have been 15 in my testing above to add up to 1min). So if I set the refresh interval to less than that, then by the time the silo is active it can already receive grain placements.

However, I assume there a reason why it defaults to the relatively long 1min, i.e. would it be a problem for me to set it to a very low value like 5 sec? Is the process to refresh type info particularly expensive/inefficient?

Also, it would be better if it refreshed immediately after becoming active in the cluster, in which case it is ok to be on a longer timer for future refreshes, at least as far as the impact on rolling upgrades are concerned. Perhaps that is the change/PR that @ReubenBond referred to in the title of this issue...
